### PR TITLE
Remove global `use_hip` flag in setup

### PR DESCRIPTION
**Pull request by kmaehashi**

### CODE REVIEW

File: install/cupy_builder/install_build.py [hunk 1]
```diff
@@ -23,8 +23,6 @@
 _rocm_path = 'NOT_INITIALIZED'
 _compiler_base_options = None
 
-use_hip = bool(int(os.environ.get('CUPY_INSTALL_USE_HIP', '0')))
```
Comment (leofang) [inline install/cupy_builder/install_build.py:26]: Just curious, this env var is no longer needed?

Comment (emcastillo) [reply install/cupy_builder/install_build.py:26]: It is present in [`context`](https://github.com/cupy/cupy/blob/master/install/cupy_builder/_context.py#L22-L23)

**Comment by emcastillo** [comment]
Seems we need to fix tests :)
```
@pytest.mark.skipif(build.use_hip, reason='For CUDA environment') |  
-- | --
00:16:41.305516 STDOUT 1592] | E   AttributeError: module 'cupy_builder.install_build' has no attribute 'use_hip'
```

File: install/cupy_builder/install_build.py (hunk 1 above)
Comment (kmaehashi) [reply install/cupy_builder/install_build.py:26]: The code was under refactoring and there were two flags for `CUPY_INSTALL_USE_HIP` (here and context).

Comment (leofang) [reply install/cupy_builder/install_build.py:26]: I see, I missed that!

**Review by emcastillo — APPROVED** [review]

**Pull request merged**